addons/controllers/csi/vspherecsiconfig_datavalues.go

```diff
@@ -22,25 +22,32 @@ type DataValuesVSpherePVCSI struct {
 }
 
 type DataValuesVSphereCSI struct {
-	TLSThumbprint         string `yaml:"tlsThumbprint"`
-	Namespace             string `yaml:"namespace"`
-	ClusterName           string `yaml:"clusterName"`
-	Server                string `yaml:"server"`
-	Datacenter            string `yaml:"datacenter"`
-	PublicNetwork         string `yaml:"publicNetwork"`
-	Username              string `yaml:"username"`
-	Password              string `yaml:"password"`
-	Region                string `yaml:"region"`
-	Zone                  string `yaml:"zone"`
-	InsecureFlag          bool   `yaml:"insecureFlag"`
-	UseTopologyCategories bool   `yaml:"useTopologyCategories"`
-	ProvisionTimeout      string `yaml:"provisionTimeout"`
-	AttachTimeout         string `yaml:"attachTimeout"`
-	ResizerTimeout        string `yaml:"resizerTimeout"`
-	VSphereVersion        string `yaml:"vSphereVersion"`
-	HTTPProxy             string `yaml:"http_proxy"`
-	HTTPSProxy            string `yaml:"https_proxy"`
-	NoProxy               string `yaml:"no_proxy"`
-	DeploymentReplicas    int32  `yaml:"deployment_replicas"`
-	WindowsSupport        bool   `yaml:"windows_support"`
+	TLSThumbprint         string                          `yaml:"tlsThumbprint"`
+	Namespace             string                          `yaml:"namespace"`
+	ClusterName           string                          `yaml:"clusterName"`
+	Server                string                          `yaml:"server"`
+	Datacenter            string                          `yaml:"datacenter"`
+	PublicNetwork         string                          `yaml:"publicNetwork"`
+	Username              string                          `yaml:"username"`
+	Password              string                          `yaml:"password"`
+	Region                string                          `yaml:"region"`
+	Zone                  string                          `yaml:"zone"`
+	InsecureFlag          bool                            `yaml:"insecureFlag"`
+	UseTopologyCategories bool                            `yaml:"useTopologyCategories"`
+	ProvisionTimeout      string                          `yaml:"provisionTimeout"`
+	AttachTimeout         string                          `yaml:"attachTimeout"`
+	ResizerTimeout        string                          `yaml:"resizerTimeout"`
+	VSphereVersion        string                          `yaml:"vSphereVersion"`
+	HTTPProxy             string                          `yaml:"http_proxy"`
+	HTTPSProxy            string                          `yaml:"https_proxy"`
+	NoProxy               string                          `yaml:"no_proxy"`
+	DeploymentReplicas    int32                           `yaml:"deployment_replicas"`
+	WindowsSupport        bool                            `yaml:"windows_support"`
+	NetPermissions        map[string]*NetPermissionConfig `yaml:"netpermissions"`
```

Comment:
wondering how should user config this? maybe similar to https://github.com/vmware-tanzu/tanzu-framework/blob/511c4d538798dbdaae886de7096652813ed5ce1a/providers/yttcb/clusterbootstrap.yaml#L88 
need to add data values ?

Reply:
yes, jeff user can provide netpermission values in similar way using VSphereCSIConfig